### PR TITLE
Jabber Plugin improvements

### PR DESCRIPTION
**Pull request by grignix**

### CODE REVIEW

**Comment by grignix** [comment]
Misspelling

**Comment by joschi** [comment]
@grignix What issues are you trying to solve or which features are you trying to add with this PR?

File: pom.xml [hunk 1]
```diff
@@ -94,20 +94,6 @@
                     <transformers>
                         <transformer implementation="org.apache.maven.plugins.shade.resource.ServicesResourceTransformer" />
                     </transformers>
-                    <relocations>
```
Comment (joschi) [inline pom.xml]: Why did you remove the relocations? They are there for a reason.

File: src/main/java/org/graylog2/alarmcallbacks/jabber/JabberAlarmCallback.java [hunk 2]
```diff
@@ -45,22 +46,23 @@
     private static final String CK_REQUIRE_SECURITY = "require_security";
     private static final String CK_RECIPIENT = "recipient";
 
-    private XMPPConnection connection;
+    private AbstractXMPPConnection connection;
```
Comment (joschi) [inline src/main/java/org/graylog2/alarmcallbacks/jabber/JabberAlarmCallback.java]: Why did you replace the interface (`XMPPConnection`) with an abstract class?

File: src/main/java/org/graylog2/alarmcallbacks/jabber/JabberAlarmCallback.java [hunk 3]
```diff
@@ -101,16 +102,23 @@ public void call(final Stream stream, final AlertCondition.CheckResult result) t
                         isNullOrEmpty(config.getString(CK_SERVICE_NAME)) ? config.getString(CK_HOSTNAME) : config.getString(CK_SERVICE_NAME)
                 );
                 throw new AlarmCallbackException("Unable to connect to XMPP server " + serverString, e);
+                } catch (InterruptedException e) {
+                        e.printStackTrace();
```
Comment (joschi) [inline src/main/java/org/graylog2/alarmcallbacks/jabber/JabberAlarmCallback.java]: Please log the exception properly.

File: src/main/java/org/graylog2/alarmcallbacks/jabber/JabberAlarmCallback.java [hunk 4]
```diff
@@ -101,16 +102,23 @@ public void call(final Stream stream, final AlertCondition.CheckResult result) t
...
         }
 
         String messageRecipient = config.getString(CK_RECIPIENT);
         String messageBody = new JabberAlarmCallbackFormatter(stream, result).toString();
-        Message message = new Message(messageRecipient, messageBody);
         try {
+        Message message = new Message(messageRecipient, messageBody);
             connection.sendStanza(message);
+            connection.disconnect();
         } catch (SmackException.NotConnectedException e) {
```
Comment (joschi) [inline src/main/java/org/graylog2/alarmcallbacks/jabber/JabberAlarmCallback.java]: Please collapse these 3 identical `catch` blocks into one `catch` block: https://docs.oracle.com/javase/7/docs/technotes/guides/language/catch-multiple.html

File: pom.xml (hunk 1 above)
Comment (grignix) [inline pom.xml]: If you compile together with relocations then the plugin is going to but does not work, the issus is already created (https://github.com/graylog-labs/graylog-plugin-jabber/issues/11) but not solved

File: src/main/java/org/graylog2/alarmcallbacks/jabber/JabberAlarmCallback.java (hunk 2 above)
Comment (grignix) [inline src/main/java/org/graylog2/alarmcallbacks/jabber/JabberAlarmCallback.java]: https://github.com/igniterealtime/Smack/wiki/Smack-4.1-Readme-and-Upgrade-Guide
"XMPPConnection is now an interface. Use either AbstractXMPPConnection or one of its subclasses as static type when declaring the connection"

File: src/main/java/org/graylog2/alarmcallbacks/jabber/JabberAlarmCallback.java (hunk 3 above)
Comment (grignix) [inline src/main/java/org/graylog2/alarmcallbacks/jabber/JabberAlarmCallback.java]: Done

File: src/main/java/org/graylog2/alarmcallbacks/jabber/JabberAlarmCallback.java (hunk 4 above)
Comment (grignix) [inline src/main/java/org/graylog2/alarmcallbacks/jabber/JabberAlarmCallback.java]: Done

**Comment by joschi** [comment]
@grignix Thanks for your contributions. I've addressed all of the issues in `master` now.
